test/int/federatedauth_test.go

```diff
@@ -0,0 +1,146 @@
+package int
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.mongodb.org/atlas/mongodbatlas"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/common"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/testutil"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/toptr"
+)
+
+var _ = Describe("AtlasFederatedAuth test", Label("AtlasFederatedAuth", "federated-auth"), func() {
+	var orgID string
+	var testNamespace *corev1.Namespace
+	var stopManager context.CancelFunc
+	var connectionSecret corev1.Secret
+
+	var originalConnectedOrgConfig *mongodbatlas.FederatedSettingsConnectedOrganization
+	var originalFederationSettings *mongodbatlas.FederatedSettings
+	var originalIdp *mongodbatlas.FederatedSettingsIdentityProvider
+
+	var fedauth *mdbv1.AtlasFederatedAuth
+
+	BeforeAll(func() {
+		It("Should verify if an IDP is configured for Test organization", func() {
+
+			By("Checking if ATLAS_ORG_ID is set", func() {
+				orgID = os.Getenv("ATLAS_ORG_ID")
+				Expect(orgID).NotTo(BeEmpty())
+			})
+
+			By("Getting original IDP", func() {
+				originalIdp, _, err := atlasClient.FederatedSettings.Get(context.Background(), orgID)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(originalIdp).NotTo(BeNil())
+			})
+
+			By("Checking if Federation Settings enabled for the org", func() {
+				originalFederationSettings, _, err := atlasClient.FederatedSettings.Get(context.Background(), orgID)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(originalFederationSettings).NotTo(BeNil())
+			})
+
+			var err error
+			By("Getting existing org config", func() {
+				originalConnectedOrgConfig, _, err = atlasClient.FederatedSettings.GetConnectedOrg(context.Background(), originalFederationSettings.ID, orgID)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(err).NotTo(BeNil())
+			})
+		})
+
+		It("Starting the operator with protection OFF", func() {
+			testNamespace, stopManager = prepareControllers(false)
```

Comment:
You need to add the setup to the new controller or this test will not work. There will be no one to handle the new spec events.